app/components/core/add-frame.tsx

```diff
@@ -52,78 +34,86 @@ export default function AddFrame({ onSave }: AddFrameProps) {
                 <Input
                   id="url"
                   name="url"
-                  defaultValue="https://www.google.com"
+                  placeholder="https://www.google.com"
                   className="col-span-2 h-8"
-                  value={frame.url}
-                  onChange={(e) => setFrame({...frame, url: e.target.value})}
+                  required
+                  pattern="https?://.*"
```

Comment:
i do minimal client side validation these days, these work and have js and css hooks if needed.